jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -497,44 +510,40 @@ void parseAndFillForContent()
 
     private int fillRequestBuffer()
     {
-        if (_retainableByteBuffer != null && _retainableByteBuffer.isRetained())
+        // We are not in a race here for the request buffer as we have not yet received a request,
+        // so there are not any possible legal threads calling #parseContent or #completed.
```

Comment:
The comment is incorrect.
This method is called from `parseAndFillForContent()`, which is called by the application either reading or demanding, so a request has indeed been received.